app/forms/FileUploadCountProvider.scala

```diff
@@ -25,18 +25,28 @@ import scala.util.Try
 
 class FileUploadCountProvider {
 
-  def apply(): Form[FileUploadCount] = Form("value" -> of(fileUploadCountFormatter("howManyFilesUpload.invalid")))
+  def apply(): Form[FileUploadCount] = Form("value" -> of(fileUploadCountFormatter()))
 
-  def fileUploadCountFormatter(errorKey: String): Formatter[FileUploadCount] = new Formatter[FileUploadCount] {
+  private def isMissing(key: String, data: Map[String, String]) =
+    data.get(key).map(_.trim) match {
+      case None | Some("") => Left(Seq(FormError(key, "howManyFilesUpload.missing")))
+      case Some(mrn)       => Right(mrn)
+    }
+
+  private def isInvalid(key: String, data: String) =
+    Try(data.toInt).toOption
+      .flatMap(FileUploadCount(_)) match {
+      case None      => Left(Seq(FormError(key, "howManyFilesUpload.invalid")))
+      case Some(mrn) => Right(mrn)
```

Comment:
Is this `mrn` or `count`, or...? Bit confused.

Reply:
Name updated :+1: